app/utils/chart-config/api.ts

```diff
@@ -33,31 +37,35 @@ export const createConfig = async (state: ConfiguratorStatePublishing) => {
 type UpdateConfigOptions = {
   key: string;
   userId: number;
+  published_state?: PUBLISHED_STATE;
 };
 
 export const updateConfig = async (
   state: ConfiguratorStatePublishing,
   options: UpdateConfigOptions
 ) => {
-  const { key, userId } = options;
+  const { key, userId, published_state } = options;
 
   return apiFetch<InferAPIResponse<typeof apiConfigUpdate, "POST">>(
     "/api/config-update",
     {
       method: "POST",
-      data: {
-        key,
-        userId,
-        data: {
+      data: omitBy(
+        {
           key,
-          version: state.version,
-          dataSource: state.dataSource,
-          layout: state.layout,
-          chartConfigs: state.chartConfigs,
-          activeChartKey: state.activeChartKey,
-          publishedState: state.publishedState,
+          userId,
+          data: {
+            key,
+            version: state.version,
+            dataSource: state.dataSource,
+            layout: state.layout,
+            chartConfigs: state.chartConfigs,
+            activeChartKey: state.activeChartKey,
+          },
+          published_state,
         },
-      },
+        isUndefined
```

Comment:
I guess this could be avoided if published_state was a required property?

Reply:
I think this will stay even if I make published_state required as we do not want each time we update a user config, to have to pass published_state.